python/tests/log.py

```diff
@@ -8,70 +8,66 @@
     default_logging_callback,
 )
 
-from test_systems import TestSystem
+from test_systems import SystemForTests
 
 
-class TestLogging(unittest.TestCase):
-    @staticmethod
-    def tearDownClass():
-        set_logging_callback(default_logging_callback)
+def setup_module(module):
+    """setup any state specific to the execution of the given module."""
 
-    def test_log_message(self):
-        recorded_events = []
 
-        def record_log_events(level, message):
-            recorded_events.append((level, message))
+def teardown_module(module):
+    set_logging_callback(default_logging_callback)
 
-        set_logging_callback(record_log_events)
 
-        calculator = DummyCalculator(
-            cutoff=3.2,
-            delta=0,
-            name="log-test-info: test info message",
-        )
-        _ = calculator.compute(TestSystem())
+def test_log_message():
```

Comment:
There also exists an option in pytest to collect the log. This is fine here but just for the future.

https://docs.pytest.org/en/6.2.x/logging.html

Reply:
yeah, the log testing options in pytest look powerful but also a bit hard to understand for non-pytest experts. Given that we only test a handful of cases, I find something more explicit to be easier to follow